From eb944186394794cdc96f11b2017d863637c42ea6 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 5 Apr 2016 13:35:14 +0300
Subject: [PATCH 1281/1383] staging: fsl-dpaa2/mac: Fix unregister_netdev issue

We only register the netdevice associated with a mac object if
ONFIG_FSL_DPAA2_MAC_NETDEV is set, but we always unregister it
during device remove(). Fix this by ifdef-ing the unregister
operation.

Also ifdef the change in netdevice name as it only makes sense
under this option.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index c86c4b1..696ddb4 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -532,7 +532,10 @@ static int dpaa2_mac_probe(struct fsl_mc_device *mc_dev)
 	priv->netdev = netdev;
 
 	SET_NETDEV_DEV(netdev, dev);
+
+#ifdef CONFIG_FSL_DPAA2_MAC_NETDEVS
 	snprintf(netdev->name, IFNAMSIZ, "mac%d", mc_dev->obj_desc.id);
+#endif
 
 	dev_set_drvdata(dev, priv);
 
@@ -682,7 +685,9 @@ static int dpaa2_mac_remove(struct fsl_mc_device *mc_dev)
 	struct device		*dev = &mc_dev->dev;
 	struct dpaa2_mac_priv	*priv = dev_get_drvdata(dev);
 
+#ifdef CONFIG_FSL_DPAA2_MAC_NETDEVS
 	unregister_netdev(priv->netdev);
+#endif
 	teardown_irqs(priv->mc_dev);
 	dpmac_close(priv->mc_dev->mc_io, 0, priv->mc_dev->mc_handle);
 	fsl_mc_portal_free(priv->mc_dev->mc_io);
-- 
2.8.1

